stcgal/protocols.py

```diff
@@ -77,7 +78,30 @@ def __init__(self, port, baud_handshake, baud_transfer):
         self.debug = False
         self.status_packet = None
         self.protocol_name = None
-
+        self.progress_cb = self.progress_bar_cb
+        self.bar = None
+        
+    def set_progress_cb(self, cb):
+        """Set a user-defined callback for progress-reporting
+        instead of default tqdm progress bar. The cb function takes 
+        three arguments: position, increment (e.g. how many bytes were written)
+        and maximum value"""
+        self.progress_cb = self.progress_bar_cb
```

Comment:
I think you want to actually use the cb argument here.

Reply:
Hm, you mean add a user-supplied arg to the callback? Okay, will do.

Reply:
No, I mean you should actually set the callback to the argument provided to the set_progress_cb() method.